File: example_solutions/javascript/test5.js
ans=p=>{var j=Array,c=t=L=R=1,d=[...j(7)].map(x=>j(6).fill(0)),g=(x,y)=>d[x]?.[y]==c
for(e of p){j=d[e].indexOf(0)
if(j<0)break
for(d[e][j]=c;g(e,j-L++););for(Q=-2;++Q<3;){if(L+R>5)return c*t
for(L=R=1;g(e-L,j-Q*L++););for(;g(e+R,j+Q*R);++R);}c=-c
t++}return 0}

File: example_solutions/C/test5.c
#define g(x,y);while(0<=x&x<7&0<=y&y<6?d[x][y]==c:0)++
long ans(*s,*e){for(int d[7][6]={},c=1,t=1;s<e;++s){int j=0,E=*s,L=1,R=1,Q=-2,*D=d[E];if(D[5])break;for(;D[j];++j)g(E,j-L)L;for(D[j]=c;++Q<3;){if(L+R>4)return c*t;L=R=1
g(E-L,j-Q*L)L
g(E+R,j+Q*R)R;}c=-c;t++;}return 0;}